gno.land/cmd/gnoland/config_edit.go

```diff
@@ -0,0 +1,264 @@
+
+type configEditCfg struct {
+	configPath string
+}
+
+// newConfigEditCmd creates the new config edit command
+func newConfigEditCmd(io commands.IO) *commands.Command {
+	cfg := &configEditCfg{}
+
+	cmd := commands.NewCommand(
+		commands.Metadata{
+			Name:       "edit",
+			ShortUsage: "config edit <key> <value>",
+			ShortHelp:  "Edits the Gno node configuration",
+			LongHelp: "Edits the Gno node configuration locally by setting " +
+				"the given value to the appropriate key",
+		},
+		cfg,
+		func(_ context.Context, args []string) error {
+			return execConfigEdit(cfg, io, args)
+		},
+	)
+
+	return cmd
+}
+
+func (c *configEditCfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.StringVar(
+		&c.configPath,
+		"config-path",
+		"./config.toml",
+		"the path to the node's config.toml",
+	)
+}
+
+func execConfigEdit(cfg *configEditCfg, io commands.IO, args []string) error {
+	// Load the config
+	loadedCfg, err := config.LoadConfigFile(cfg.configPath)
+	if err != nil {
+		return fmt.Errorf("unable to load config, %w", err)
+	}
+
+	// Make sure the edit arguments are valid
+	if len(args) != 2 {
+		return errInvalidConfigEditArgs
+	}
+
+	// Get the config value using reflect
+	configValue := reflect.ValueOf(loadedCfg).Elem()
+
+	var (
+		key   = args[0]
+		value = args[1]
+	)
+
+	// Update the config field
+	if err := updateFieldAtPath(
+		configValue,
+		key,
+		value,
+	); err != nil {
+		return fmt.Errorf("unable to update config field, %w", err)
+	}
+
+	// Make sure the config is now valid
+	if err := loadedCfg.ValidateBasic(); err != nil {
+		return fmt.Errorf("unable to validate config, %w", err)
+	}
+
+	// Save the config
+	if err := config.WriteConfigFile(cfg.configPath, loadedCfg); err != nil {
+		return fmt.Errorf("unable to save updated config, %w", err)
+	}
+
+	io.Printfln("Updated configuration saved at %s", cfg.configPath)
+
+	return nil
+}
+
+// updateFieldAtPath updates the field at the given path, with the given value
+func updateFieldAtPath(currentValue reflect.Value, path string, value string) error {
+	// Get the value path, with sections separated out by a period
+	pathParts := strings.Split(path, ".")
+
+	// Look at the current section, and figure out if
+	// it's a part of the current struct
+	field := currentValue.FieldByName(pathParts[0])
+	if !field.IsValid() {
+		return fmt.Errorf("invalid field, %s", pathParts[0])
+	}
+
+	// Dereference the field if needed
+	if field.Kind() == reflect.Ptr {
+		field = field.Elem()
+	}
+
+	// Check if this is not the end of the path
+	// ex: x.y.field
+	if len(pathParts) > 1 {
+		// Get the remaining path
+		remainingPath := strings.Join(pathParts[1:], ".")
+
+		// Recursively try to traverse the path and update the given field
+		return updateFieldAtPath(field, remainingPath, value)
+	}
+
+	// We've reached the actual field, check if it can be updated
+	if !field.CanSet() {
+		return fmt.Errorf("unable to set value")
+	}
```

Comment:
When can this happen?

Reply:
I put it as a safeguard for modifying unexported fields in the underlying struct using reflect:
https://pkg.go.dev/reflect#Value.CanSet

You'd get this error if you wanted to modify a cache value like `chainID` on the `BaseConfig`:
https://github.com/zivkovicmilos/gno/blob/fb2b385c48cb706546f3d0d4599fc953ea400dee/tm2/pkg/bft/config/config.go#L223-L224

`gnoland config edit chainID morgan`

Reply:
ah yes, right. I forgot about unexported :P